client/src/ui/organisms/placement/index.tsx

```diff
@@ -191,94 +209,61 @@ export function PlacementInner({
   renderer,
+  typ,
 }: {
-  pong: PlacementData;
+  pong?: PlacementData;
   extraClassNames?: string[];
   cta?: string;
   imageWidth?: number;
   imageHeight?: number;
   style?: object;
   renderer: (PlacementRenderArgs) => JSX.Element;
+  typ: string;
 }) {
   const isServer = useIsServer();
   const user = useUserData();
   const isVisible = usePageVisibility();
   const gleanClick = useGleanClick();
 
-  const observer = useRef<IntersectionObserver | null>(null);
-  const timer = useRef<Timer>({ timeout: null, start: null });
-  const place = useCallback(
-    (node) => {
-      if (pong && node !== null && !observer.current) {
-        const observerOptions = {
-          root: null,
-          rootMargin: "0px",
-          threshold: [0.5],
-        };
-        const intersectionObserver = new IntersectionObserver((entries) => {
-          const [{ isIntersecting = false, intersectionRatio = 0 } = {}] =
-            entries;
-          if (isIntersecting && intersectionRatio >= 0.5) {
-            if (timer.current.timeout === null) {
-              timer.current = {
-                timeout: window?.setTimeout?.(() => {
-                  viewed(pong, observer?.current);
-                  gleanClick("pong: pong->viewed");
-                  timer.current = { timeout: -1, start: -1 };
-                }, 1000),
-                start: Date.now(),
-              };
-            }
-          } else if (
-            !isIntersecting &&
-            intersectionRatio <= 0.5 &&
-            timer.current.timeout !== null
-          ) {
-            clearTimeout(timer.current.timeout);
-            timer.current = { timeout: null, start: null };
-          }
-        }, observerOptions);
-        observer.current = intersectionObserver;
-        intersectionObserver.observe(node);
-      }
-    },
-    [pong, gleanClick]
-  );
+  const timer = useRef<Timer>({ timeout: null });
 
-  const { image, copy } = pong?.fallback || pong || {};
-  const { click } = pong || {};
-  useEffect(() => {
-    return () => observer.current?.disconnect();
+  const [node, setNode] = useState<HTMLElement>();
+  const isIntersecting = useIsIntersecting(node, INTERSECTION_OPTIONS);
+
+  const sendViewed = useCallback(() => {
+    viewed(pong);
+    gleanClick(`pong: pong->viewed ${typ}`);
+    timer.current = { timeout: -1 };
+  }, [pong, gleanClick, typ]);
+
+  const place = useCallback((node: HTMLElement | null) => {
+    if (node) {
+      setNode(node);
+    }
   }, []);
 
   useEffect(() => {
     if (timer.current.timeout !== -1) {
-      // timeout !== -1 means the viewed has been sent
-      if (!isVisible && timer.current.timeout !== null) {
+      // timeout !== -1 means the viewed has not been sent
+      if (isVisible && isIntersecting) {
+        if (timer.current.timeout === null) {
+          timer.current = {
+            timeout: window.setTimeout(sendViewed, 1000),
+          };
+        }
+      } else if (timer.current.timeout !== null) {
```

Comment:
nit: (again in my own code), it might be "more reacty" to clear the timeout in the cleanup function (which gets run between state changes), and ensures we don't have any dangling timeout if the component unmounts:
```js
  useEffect(() => {
    if (timer.current.timeout !== -1) {
      // timeout !== -1 means the viewed has not been sent
      if (isVisible && isIntersecting) {
        if (timer.current.timeout === null) {
          timer.current = {
            timeout: window.setTimeout(sendViewed, 1000),
          };
        }
      }
    }
    return () => {
      if (timer.current.timeout !== null && timer.current.timeout !== -1) {
        clearTimeout(timer.current.timeout);
        timer.current = { timeout: null };
      }
    }
  }, [isVisible, isIntersecting, sendViewed]);
```

Reply:
👍